qemu/tests/block_hotplug.py

```diff
@@ -0,0 +1,168 @@
+        """
+        image_params = params.object_params(image_name)
+        o = storage.get_image_filename(image_params, data_dir.get_data_dir())
+        return o
+
+    def find_disk(session, cmd):
+        """
+        Find all disks in guest.
+        """
+        if params.get("os_type") == "linux":
+            pattern = params.get("get_disk_pattern", "^/dev/vd[a-z]*$")
+        else:
+            pattern = "^\d+"
+            cmd = params.get("get_disk_index", "wmic diskdrive get index")
+
+        output = session.cmd_output_safe(cmd)
+        disks = re.findall(pattern, output, re.M)
+        return disks
+
+    def get_new_disk(disk1, disk2):
+        """
+        Get the different disk between disk1 and disk2.
+        """
+        disk = list(set(disk2).difference(set(disk1)))
+        return disk
+
+    img_list = params.get("images").split()
+    img_format_type = params.get("img_format_type", "qcow2")
+    pci_type = params.get("pci_type", "virtio-blk-pci")
+    pause = float(params.get("virtio_block_pause", 5.0))
+    blk_num = int(params.get("blk_num", 1))
+    repeat_times = int(params.get("repeat_times", 3))
+    timeout = int(params.get("login_timeout", 360))
+    disk_op_timeout = int(params.get("disk_op_timeout", 360))
+    get_disk_cmd = params.get("get_disk_cmd")
+    context_msg = "Running sub test '%s' %s"
+    device_list = []
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    for i in xrange(repeat_times):
+        error_context.context("Hotplug block device (iteration %d)" % i,
+                              logging.info)
+
+        session = vm.wait_for_login(timeout=timeout)
+
+        sub_type = params.get("sub_type_before_plug")
+        if sub_type:
+            error_context.context(context_msg % (sub_type, "before hotplug"),
+                                  logging.info)
+            utils_test.run_virt_sub_test(test, params, env, sub_type)
+
+        for num in xrange(blk_num):
+            device = qdevices.QDevice(pci_type)
+            if params.get("need_plug") == "yes":
+                if params.get("need_controller", "no") == "yes":
+                    controller_model = params.get("controller_model")
+                    controller = qdevices.QDevice(controller_model)
+                    controller.hotplug(vm.monitor)
+                    ver_out = controller.verify_hotplug("", vm.monitor)
+                    if not ver_out:
+                        err = "%s is not in qtree after hotplug" % controller_model
+                        raise exceptions.TestFail(err)
+
+                disks_before_plug = find_disk(session, get_disk_cmd)
+
+                drive = qdevices.QRHDrive("block%d" % num)
+                drive.set_param("file", find_image(img_list[num + 1]))
+                drive.set_param("format", img_format_type)
+                drive_id = drive.get_param("id")
+                drive.hotplug(vm.monitor)
+                time.sleep(pause)
```

Comment:
Please test remove  sleep in line#103 and line#108 , then add sleep in line#131.